requirements_lint.txt

```diff
@@ -0,0 +1,7 @@
+flake8==3.9.2
```

Comment:
I don't think you need these unless you wanted to install lintlizard with `--no-deps` (not recommended), since lintlizard is supposed to be where you manage linter versions. Only lintlizard itself and any project-specific extras (none so far for this one) should be in this file.

Reply:
This is a point of contention in Python packaging world.

The reason to pin these is so that, if your dependency (like LintLizard) doesn't pin versions, you don't end up with accidentally breaking your future build.

The usual set-up is to have actual dependencies for your project listed in requirements*.in files (or in setup.py directly when you don't need multiple "environments"), and then pip-compile them into .txt files that still have the exact versions for every dependency.

I'd like to try to improve our requirements-setup in the follow-up branch if I can find the time (I need to look into release-rabbit too).

Reply:
> The reason to pin these is so that, if your dependency (like LintLizard) doesn't pin versions, you don't end up with accidentally breaking your future build.

This is a perfect reason in general case. However we're specifically doing it differently for LintLizard.  LintLizard pins _its_ dependencies exactly. The intention is to avoid having each repo pinning a bunch of tiny flake8 plugins and invariably ending up with a different set of versions. All version management for all linting tools and plugins for them should happen in LintLizard instead. The end result is you'll have just one thing to update across 10-20 repos instead of ~10-20 things across each of 10-20 (which adds up to a non-trivial maintenance burden).

Reply:
Sure: I've created https://github.com/closeio/socketshark/pull/84 since this one is already merged.